src/server/analytic_services/prometheus_reports/noobaa_endpoint_report.js

```diff
@@ -192,6 +257,15 @@ class NooBaaEndpointReport extends BasePrometheusReport {
         metric.prom_instance.inc(labels, value);
     }
 
+    // Set value metric
+    set(name, labels, value) {
+        if (!this._metrics) return;
+        const metric = this._metrics[name];
+        if (!metric) throw new Error(`Unknown metric ${name}`);
+        if (metric.type !== 'Gauge') throw new Error(`Metric ${name} is not Gauge`);
+        if (metric.collect) metric.prom_instance.collect = metric.collect(metric.prom_instance, labels, value);
```

Comment:
looking at the example [here](https://github.com/siimon/prom-client#custom-metrics), the use of `collect` function seems incorrect.
IIUC, the collect function will be called when Prometheus scrapes the metrics, and this is when the average should be calculated (not every  10 seconds when a sample is taken). I just could not understand how labels are working together with a `collect` function